src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -328,6 +325,14 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
                 return retVal;
             }
 
+            if (CurlPoll.CURL_POLL_REMOVE == socketAction)
```

Comment:
So if socketAction is CurlPoll.CURL_POLL_REMOVE, we first remove it here, and then we add it back below as part of isAdd?